src/Microsoft.Identity.Web.AgentIdentities/AgentIdentitiesExtension.cs

```diff
@@ -79,6 +80,35 @@ public static AuthorizationHeaderProviderOptions WithAgentUserIdentity(this Auth
             return options;
         }
 
+        /// <summary>
+        /// Updates the options to acquire a token for the agent user identity using the user's object id (OID).
+        /// </summary>
+        /// <param name="options">Authorization header provider options.</param>
+        /// <param name="agentApplicationId">The agent identity application (client) ID.</param>
+        /// <param name="userId">The user's object id (OID).</param>
+        /// <returns>The updated authorization header provider options (in place; not a clone).</returns>
+        /// <remarks>
+        /// If both a UPN and an OID are present in the options (not expected via the public API), UPN takes precedence.
+        /// </remarks>
+        public static AuthorizationHeaderProviderOptions WithAgentUserIdentity(this AuthorizationHeaderProviderOptions options, string agentApplicationId, Guid userId)
```

Comment:
naming: I would name it `userObjectId` to be consistent with the portal

<img width="843" height="63" alt="image" src="https://github.com/user-attachments/assets/251bc2b1-2ae3-4f6b-90db-b36d92d876c5" />